File: Design-Patterns-Using-Java/Basics OOPS/AssociationRelationship.java
import java.util.ArrayList;
import java.util.List;

class Student {
    private String name;

    public Student(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Student: " + name;
    }
}

class SchoolClass {
    private String className;
    private List<Student> students;

    public SchoolClass(String className) {
        this.className = className;
        this.students = new ArrayList<>();
    }

    public void addStudent(Student student) {
        students.add(student);
    }

    public void display() {
        System.out.println("Students in class: " + className);        
        for (Student student : students) {
            System.out.println(student);
        }
    }
}

public class AssociationRelationship {
    public static void main(String[] args) {
        SchoolClass schoolClass = new SchoolClass("S1");
        Student student1 = new Student("John Smith");
        Student student2 = new Student("Jane Smith");

        schoolClass.addStudent(student1);
        schoolClass.addStudent(student2);

        schoolClass.display();
    }
}

/*
Students in class: S1
Student: John Smith
Student: Jane Smith
*/

File: Design-Patterns-Using-CSharp/Basics OOPS/AssociationRelationship.cs
using System;
using System.Collections.Generic;

class Student
{
    private string name;

    public Student(string name)
    {
        this.name = name;
    }

    public override string ToString()
    {
        return "Student: " + name;
    }
}

class SchoolClass
{
    private string className;
    private List<Student> students;

    public SchoolClass(string className)
    {
        this.className = className;
        this.students = new List<Student>();
    }

    public void AddStudent(Student student)
    {
        students.Add(student);
    }

    public void Display()
    {
        foreach (Student student in students)
        {
            Console.WriteLine(student);
        }
    }
}

public class AssociationRelationship
{
    public static void Main(string[] args)
    {
        SchoolClass schoolClass = new SchoolClass("SS1");
        Student student1 = new Student("John Smith");
        Student student2 = new Student("Jane Smith");

        schoolClass.AddStudent(student1);
        schoolClass.AddStudent(student2);

        schoolClass.Display();
    }
}
